SERVERPLUGINS.md

```diff
@@ -3,36 +3,338 @@
+/usr/local/lib/node_modules/my-signalk-plugin -> /home/me/dev/my-signalk-plugin
+```
 
-See [Ais Reporter](https://github.com/SignalK/aisreporter/issues) for an example.
+Then from your SignalK server's configuration directory (`$HOME/.signalk/` by default), you link your plugin.
 
-You can make sections of your schema collapsible.
+```
+$ cd ~/.signalk
+$ npm link my-signalk-plugin
+...
 
-For example, to make all data in an object called 'myObject' collapsible:
+/home/me/.signalk/node_modules/my-signalk-plugin -> /usr/local/lib/node_modules/my-signalk-plugin -> /home/me/dev/my-signalk-plugin
 ```
+
+### Plugin skeleton
+
+In your module directory, create `index.js` with the following content:
+
+```javascript
+module.exports = function (app) {
+  var plugin = {};
+
+  plugin.id = 'my-signalk-plugin';
+  plugin.name = 'My Great Plugin';
+  plugin.description = 'Plugin that does stuff';
+
+  plugin.start = function (options, restartPlugin) {
+    // Here we put our plugin logic
+    app.debug('Plugin started');
+  };
+
+  plugin.stop = function () {
+    // Here we put logic we need when the plugin stops
+    app.debug('Plugin stopped');
+  };
+
+  plugin.schema = {
+    // The plugin schema
+  };
+
+  return plugin;
+};
+
+```
+
+You should be able to restart the server now. If you set the `DEBUG` environment variable, you can get debug output about the plugin loading process:
+
+```
+$ DEBUG=signalk:interfaces:plugins signalk-server
+```
+
+And you should see something like:
+```
+signalk:interfaces:plugins Registering plugin my-signalk-plugin +0ms
+signalk:interfaces:plugins Could not find options for plugin my-signalk-plugin, returning empty options:  +2ms
+```
+
+After everything loads well, you can debug your own plugin by running:
+
+```
+$ DEBUG=my-signalk-plugin signalk-server
+```
+
+You should be able to now enable your plugin in the admin UI (although it doesn't do anything yet).
+
+You should see something like this in your console:
+```
+my-signalk-plugin Plugin stopped +0ms
+my-signalk-plugin Plugin started +2ms
+```
+
+For development purposes, it's often nice to have some mocked data. SignalK comes with a synthesized NMEA2000 data set that can be used as sample data. You can enable this by adding `--sample-n2k-data` to the command line:
+
+```
+$ DEBUG=my-signalk-plugin signalk-server --sample-n2k-data
+```
+
+### Deltas
+
+Most of the time your plugin wants to either process data that the server gathers (e.g. data that is coming from a NMEA 2000 bus or another source) or generate data for the server to consume (e.g. data you capture from a device or the internet). In both cases you'll often work with *deltas*; data structures that describe a change in SignalK's internal data model. Note that deltas contain absolute values and not a change in the value itself, but denote a change in the data model.
+
+Deltas typically look like this:
```

Comment:
Hum. This is actually a node server internal construction and a Signal K delta. To avoid confusion we need to call it something else. My word for it is _denormalized delta_, which is a monster of s word, so I am open for suggestions.

See the SK spec for delta documentation.

Reply:
I've added some language. Would that work?